aws/modules/karpenter-nodepool/main.tf

```
@@ -15,9 +15,19 @@ resource "terraform_data" "destroyer" {
 
     command     = <<-EOF
       set -euo pipefail
-      nodeclaims=$(kubectl --kubeconfig <(echo "$${KUBECONFIG_DATA}") get nodeclaims -l "karpenter.sh/nodepool=$${NODEPOOL_NAME}" -o name)
+
+      if [ -z "$${KUBECONFIG_DATA}" ]; then
+        echo "Error: KUBECONFIG_DATA is empty"
+        exit 1
+      fi
+
+      kubeconfig_file=$(mktemp)
+      trap "rm -f '$${kubeconfig_file}' EXIT
```

Close, but you're missing the closing double quote.

```suggestion
      trap "rm -f '$${kubeconfig_file}'" EXIT
```

---

oops, thanks for this.